src/lib/Web3Function.ts

```diff
@@ -88,10 +104,72 @@ export class Web3Function {
     }
   }
 
-  private async _run(ctxData: Web3FunctionContextData) {
-    if (!this._onRun)
+  private async _invokeOnRun(ctxData: Web3FunctionOnRunContextData) {
+    const context = this._context(ctxData);
+
+    if (ctxData.operation === "onRun" && !this._onRun)
```

Comment:
is `ctxData.operation === "onRun"` required? when I just consider this function alone it should be
```Typescript
ctxData.operation !== "onRun" || !this._onRun
```